allennlp/nn/decoding/decoder_step.py

```diff
@@ -22,19 +22,28 @@ class DecoderStep(torch.nn.Module, Generic[StateType]):
     """
     def take_step(self,
                   state: StateType,
-                  allowed_actions: Set = None) -> Generator[StateType, None, None]:
+                  allowed_actions: List[Set] = None) -> Generator[StateType, None, None]:
```

Comment:
Maybe add an `Optional[]` here and in the docstring, and explain why this is the case.

Reply:
I'm not sure that explicit `Optional` annotations really add anything - you already have the `= None` there, and `Optional` is redundant with that.  But I updated the docstring.